### PR TITLE
[image-builder] Enable TLS in workspace clusters

### PR DESCRIPTION
**Pull request by WVerlaek**
## Description

Enable TLS on the `image-builder-mk3` service, in workspace clusters only. See https://github.com/gitpod-io/ops/issues/6905#issuecomment-1326586653 for more context behind this change.

Opted to keep `insecure` connections for image-builder in **application** clusters, because:
- The image-builder-mk3 NetworkPolicy was only changed in workspace clusters to allow all ingress.
- Workspace components should get removed soon from the application cluster once image builds move to workspace clusters, did not want to make server/application cluster changes which would get removed soon again.
  - Also simplifies the rollout of this change, as it's now a workspace-only change.

No need to mount image-builder TLS certs on `server`, as server proxies through `ws-manager` to call the image-builder api in workspace clusters.

## Related Issue(s)

Relates to https://github.com/gitpod-io/ops/issues/6905

## How to test

Did the following tests:

Ephemeral cluster:
- Create a goverened `workspace` cluster. Register it with an application cluster.
- Check image-builder TLS is enabled inside the workspace cluster (TLS certs are mounted)
- Enable the `movedImageBuilder` ConfigCat feature flag for your user
- Check image builds work by starting a workspace: https://gitpod.io/#imagebuild/github.com/gitpod-io/gitpod

For single-cluster installs:
- Create a preview environment, or a `workspace-preview` cluster with the `-f full` flavour.
- Check that image builds still work by starting a workspace

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=workspace
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: install/installer/pkg/components/image-builder-mk3/tlssecret.go
Comment (jenting) [inline install/installer/pkg/components/image-builder-mk3/tlssecret.go:1]:
```suggestion
// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
```

File: install/installer/pkg/components/ws-manager/configmap.go
Comment (jenting) [inline install/installer/pkg/components/ws-manager/configmap.go:166]:
```suggestion
	var imageBuilderTLS struct {
```

File: install/installer/pkg/components/ws-manager/configmap.go
Comment (jenting) [inline install/installer/pkg/components/ws-manager/configmap.go:175]:
```suggestion
		imageBuilderTLS = struct {
```

File: install/installer/pkg/components/ws-manager/configmap.go
Comment (jenting) [inline install/installer/pkg/components/ws-manager/configmap.go:263]:
```suggestion
			TLS:        imageBuilderTLS,
```

File: install/installer/pkg/components/ws-manager/configmap.go
Comment (jenting) [inline install/installer/pkg/components/ws-manager/configmap.go:182]: nit
```suggestion
			CA:          fmt.Sprintf("/%s/ca.crt", imagebuildermk3.VolumeTLSCerts),
			Certificate: fmt.Sprintf("/%s/tls.crt", imagebuildermk3.VolumeTLSCerts),
			PrivateKey:  fmt.Sprintf("/%s/tls.key", imagebuildermk3.VolumeTLSCerts),
```

File: install/installer/pkg/components/ws-manager/deployment.go
Comment (jenting) [inline install/installer/pkg/components/ws-manager/deployment.go:42]: nit:
```suggestion
			MountPath: fmt.Sprintf("/%s", imagebuildermk3.VolumeTLSCerts),
```

File: install/installer/pkg/components/image-builder-mk3/tlssecret.go
Comment (jenting) [inline install/installer/pkg/components/image-builder-mk3/tlssecret.go:31]: Is this dedicated to the Harvester preview environment?

**Comment by jenting** [comment]
I follow the PR description for the How to test:
- I create an ephemeral cluster [job](https://werft.gitpod-io-dev.com/job/ops-workspace-cluster-new-main.273), but it the image-builder-mk3 and ws-manager does not include TLS. 🤔 
- I create a workspace-preview
  - with full flavor, the image-builder-mk3 and ws-manager do not include TLS. 🤔 
  - with workspace flavor, the image-builder-mk3 and ws-amanger include TLS.

Is there anything I did wrong? @WVerlaek

**Review by jenting — APPROVED** [review]
Code LGTM

/hold
because I have questions about the PR description.

**Comment by kylos101** [comment]
> because I have questions about the PR description.

@jenting there should definitely be TLS between image-builder-mk3 and ws-manager in the workspace cluster. When you ran your test, did you add yourself to configcat and enable `movedImageBuilder`?

**Comment by kylos101** [comment]
Image build works

```
NAME                                              READY   STATUS    RESTARTS   AGE     IP               NODE                                              NOMINATED NODE   READINESS GATES
imagebuild-87601749-2801-409b-90c7-6ffc9aed5f65   1/1     Running   0          2m32s   192.168.23.197   headless-ws-ephemeral-jt-pool-ghkq                <none>           <none>
```

image-builder-mk3 pod volumes contain wsman-tls-certs
```
Volumes:
  configuration:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      image-builder-mk3-config
    Optional:  false
  wsman-tls-certs:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  ws-manager-client-tls
    Optional:    false
  pull-secret:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  workspace-registry-pull-secret
    Optional:    false
  gitpod-ca-certificate:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     
    SizeLimit:  <unset>
  cacerts:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     
    SizeLimit:  <unset>
  image-builder-mk3-tls-certs:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  image-builder-mk3-tls
    Optional:    false
  kube-api-access-xkf5g:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
```

ws-manager pod volumes contains image-builder-mk3-tls-certs:
```
Volumes:
  config:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      ws-manager
    Optional:  false
  workspace-template:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      workspace-templates
    Optional:  false
  ws-daemon-tls-certs:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  ws-daemon-tls
    Optional:    false
  tls-certs:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  ws-manager-tls
    Optional:    false
  image-builder-mk3-tls-certs:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  image-builder-mk3-tls
    Optional:    false
  storage-volume:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  ws-sync-gcloud
    Optional:    false
  kube-api-access-bfjvp:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
```

@jenting what were you checking to conclude that ephemeral workspace cluster is lacking TLS?

File: install/installer/pkg/components/image-builder-mk3/tlssecret.go
Comment (kylos101) [reply install/installer/pkg/components/image-builder-mk3/tlssecret.go:31]: Yes, `ws-dev` is for harvester previews

Context:
https://github.com/gitpod-io/gitpod/blob/4aacec9e264892232d76aba227f02e6f33da13f5/dev/preview/infrastructure/harvester/dns.tf#L32

https://github.com/gitpod-io/gitpod/blob/4aacec9e264892232d76aba227f02e6f33da13f5/dev/preview/infrastructure/harvester/cert.tf#L19

**Review by kylos101 — COMMENTED** [review]
Nice work, @WVerlaek ! A couple nits from @jenting , but nothing blocking.

**Comment by kylos101** [comment]
@aledbf could we ask for your review? It says you're an explicit code owner, this is for `gen80`. I plan to build the image, test, and start the shift Tuesday.

**Comment by kylos101** [comment]
@gitpod-io/engineering-delivery-operations-experience hey there, could we ask you for a review? This is a low risk change which we'd only be exposing to the workspace team over the holiday season.

**Review by aledbf — APPROVED** [review]

**Review by Pothulapati — APPROVED** [review]
:shipit:. Config side of things look good, and can see that the the [output file has all the things when `kind==workspace`](https://github.com/gitpod-io/gitpod/blob/5023d9d5fb05bb26253b7c47cb5fb31f32f82c41/install/installer/cmd/testdata/render/kind-workspace/output.golden)

**Pull request merged**
